### PR TITLE
Map the indices of 1p1h and 2p2h configurations to an arbitrary set of wires

### PR DESCRIPTION
**Pull request by agran2018**
**Context:** Part of the overall wire management refactor.
The new qchem templates were recently  generalized by Maria in order to allow the user to specify an arbitrary set of qubits (wires) that the templates act on. This generalization requires the templates to know explicitly all the wires involved in the single- and double-excitation configurations entering the UCCSD ansatz. 

**Description of the Change:**
The new function ``qchem.ph_pphh_wires`` has been implemented to map the particle-hole representations ``ph`` and ``pphh``, generated by ``qchem.sd_excitations``, to the wires that the qchem templates act on.

```
**Example**
n_electrons = 2
n_spinorbitals = 4
ph_confs, pphh_confs = sd_excitations(n_electrons, n_spinorbitals)
print(ph_confs)
print(pphh_confs)
[[0, 2], [1, 3]]
[[0, 1, 2, 3]]

wires=['a0', 'b1', 'c2', 'd3']
ph, pphh = ph_pphh_wires(ph_confs, pphh_confs, wires=wires)
print(ph)
print(pphh)
[['a0', 'b1', 'c2'], ['b1', 'c2', 'd3']]
[[['a0', 'b1'], ['c2', 'd3']]]
```
**Benefits:** This implementation enables compliance with the generalized PennyLane templates.

**Possible Drawbacks:** None

**Related GitHub Issues:** None

### CODE REVIEW

**Comment by agran2018** [comment]
This PR came out of the discussion with Maria about the generalization of the qchem templates.
Thanks for reviewing it.

File: qchem/pennylane_qchem/qchem.py [hunk 1]
```diff
@@ -690,6 +690,94 @@ def hf_state(n_electrons, m_spin_orbitals):
     return np.array(hf_state_on)
 
 
+def ph_pphh_wires(ph_confs, pphh_confs, wires=None):
```
Comment (josh146) [inline qchem/pennylane_qchem/qchem.py:693]: @agran2018 I have to admit, I find both the function and the argument names a bit obfuscated 🤔

File: qchem/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,30 @@
...
+  [[0, 2], [1, 3]]
+  [[0, 1, 2, 3]]
+
+  >>> wires=['a0', 'b1', 'c2', 'd3']
+  >>> ph, pphh = ph_pphh_wires(ph_confs, pphh_confs, wires=wires)
+  >>> print(ph)
+  >>> print(pphh)
+  [['a0', 'b1', 'c2'], ['b1', 'c2', 'd3']]
+  [[['a0', 'b1'], ['c2', 'd3']]]
+  ```
```
Comment (josh146) [inline qchem/CHANGELOG.md:28]:
```suggestion
  ```

```

File: qchem/pennylane_qchem/qchem.py (hunk 1 above)
Comment (agran2018) [reply qchem/pennylane_qchem/qchem.py:693]: Yes, I agree. What about this: ``convert_ph_to_wires(ph, pphh, wires=None)`` ?

Comment (mariaschuld) [reply qchem/pennylane_qchem/qchem.py:693]: Or ``particle_hole_wires``? Or ``excitations_to_wires``?

File: qchem/pennylane_qchem/qchem.py [hunk 3]
```diff
@@ -690,6 +690,94 @@ def hf_state(n_electrons, m_spin_orbitals):
...
+
+    r"""Map the indices representing the particle-hole configurations
+    generated by the Coupled-Cluster excitation operator to the wires that
+    the Unitary Coupled-Cluster Singles and Doubles (UCCSD) template will act on.
```
Comment (mariaschuld) [inline qchem/pennylane_qchem/qchem.py:697]:
```suggestion
    can be fed into the Unitary Coupled-Cluster Singles and Doubles (UCCSD) template.
```

Not sure if this is better?

File: qchem/tests/test_ph_pphh_wires.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import os
+
+import numpy as np
+import pytest
+
+import pennylane as qml
+from pennylane import qchem
+from pennylane.templates.subroutines import UCCSD
+
+
```
Comment (mariaschuld) [inline qchem/tests/test_ph_pphh_wires.py:10]: I really like your tests, they are always so explicit!

**Comment by agran2018** [comment]
> Looking good @agran2018 ! Thanks for being so flexible on this one.

Thanks to you for the generalisation of the templates and nice discussion.

File: qchem/CHANGELOG.md [hunk 5]
```diff
@@ -19,6 +19,30 @@ This release contains contributions from (in alphabetical order):
 
 <h3>New features since last release</h3>
 
+* The new function ``excitations_to_wires`` has been implemented to map the particle-hole
```
Comment (josh146) [inline qchem/CHANGELOG.md:22]: @agran2018: now that v0.10 has been released, you will need to move this entry to the top section of the changelog :)

Comment (agran2018) [reply qchem/CHANGELOG.md:5]: Done, thanks @josh146 !.

**Pull request merged**